### PR TITLE
Added Martian Wallet support

### PR DESCRIPTION
**Pull request by MartianAbhishek**
### Description
Added Martian Wallet support
### Test Plan

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/3979)

### CODE REVIEW

File: ecosystem/platform/server/app/javascript/controllers/connect_wallet_controller.ts [hunk 1]
```diff
@@ -21,24 +21,27 @@ export default class extends Controller<HTMLFormElement> {
...
+    if ("aptos" in window) {
+      return "petra";
+    } else if ("martian" in window) {
+      return "martian";
     } else if (false) {
       // TODO: Add more wallet detection logic here.
-    } else {
-      throw 'Aptos wallet not detected.';
+    }
+    {
```
Comment (zacharydenton) [inline ecosystem/platform/server/app/javascript/controllers/connect_wallet_controller.ts:37]: Either restore the `else` or remove the unnecessary block statement.

**Review by zacharydenton — COMMENTED** [review]
Is this tested? You will need to add `martian` to the list of valid wallets here: https://github.com/aptos-labs/aptos-core/blob/4025bae47a4003568f3b006849d50eac0177b8a5/ecosystem/platform/server/app/models/wallet.rb#L8

**Comment by MartianSiddharth** [comment]
For testing, we're facing this issue related to environment variables:
Screen Shot 2022-09-08 at 2 36 06 PM

**Comment by zacharydenton** [comment]
You would need to publish the module and set the environment variables for address/private key in `.env`. I tried testing your change and everything seems to work except submitting the transaction. I see this error inside the extension:
image
Is it perhaps just an issue with the way I've set up the wallet?

**Comment by MartianSiddharth** [comment]
Facing this issue while trying to sign up from Google (as part of the first step)
Screen Shot 2022-09-08 at 3 56 33 PM

Is there a way to bypass this?
Also the third step doesn't seem to have been implemented yet, how should we test it?

**Comment by zacharydenton** [comment]
I don't see any other error messages but I do note that the transaction details are stuck in a loading state in the "Approve Request" screen.
image
The other thing I notice is that after clicking "Claim NFT", the wallet briefly displays the password input before immediately switching to the "Approve Request" screen. (I'm thinking this is all an issue with the way my wallet is set up.)

**Comment by MartianAbhishek** [comment]
> I don't see any other error messages but I do note that the transaction details are stuck in a loading state in the "Approve Request" screen. image The other thing I notice is that after clicking "Claim NFT", the wallet briefly displays the password input before immediately switching to the "Approve Request" screen. (I'm thinking this is all an issue with the way my wallet is set up.)

seems like we have to use generateSignAndSubmitTransaction instead of signAndSubmitTransaction for martian

**Comment by zacharydenton** [comment]
Getting closer! Now I see this "Invalid Payload" error:

image

For reference, here is the transaction it is trying to send:

image

```js
{"type":"entry_function_payload","function":"0x5a1df0cd5a5966fc5039b390ee7d66e7fb43d05589ed529dbc56115ddcad3f0b::NFTR::claim_mint","arguments":["0x5a1df0cd5a5966fc5039b390ee7d66e7fb43d05589ed529dbc56115ddcad3f0b::NFTR::Anchor!0x384e61cd818b2f74a5bbd1ea7a53f633f0926eab35e5d15a08e9e5e902b2a8d9!0",[166,217,186,120,188,43,16,48,103,39,111,33,96,249,109,118,181,116,158,137,95,189,79,13,200,73,78,214,215,255,27,183,53,11,43,147,179,130,239,167,113,103,209,154,26,169,208,219,16,26,159,221,92,122,233,1,159,167,149,210,121,40,33,5]],"type_arguments":[]}
```

**Comment by MartianSiddharth** [comment]
@zacharydenton what network is this move module deployed to? Did you change the network in the martian wallet before signing this transaction.
To switch the network, you need to do this:

https://user-images.githubusercontent.com/99449587/189426842-a25a7af9-331e-4903-977e-4e390a204e8b.mov

**Pull request merged**